### PR TITLE
going language first for docs

### PR DESCRIPTION
**Pull request by cartersocha**
* Reducing readme size by eliminating service table and adding new file
* Updating to language first for trace docs

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ your backend as well.
 
 ## Architecture
 
-**Online Boutique** is composed of 10 microservices written in different
+**Online Boutique** is composed of 11 microservices written in different
 languages that talk to each other over gRPC. Plus one Load Generator which uses
 Locust to fake user traffic.
```
Comment (reyang) [inline README.md:127]:
```suggestion
[Locust](https://locust.io/) to fake user traffic.
```

File: docs/service_table.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+| [emailservice](../src/emailservice/README.md)                   | Ruby        | Sends users an order confirmation email (mock).                                                                                   |
+| [checkoutservice](../src/checkoutservice/README.md)             | Go            | Retrieves user cart, prepares order and orchestrates the payment, shipping and the email notification.                            |
+| [recommendationservice](../src/recommendationservice/README.md) | Python        | Recommends other products based on what's given in the cart.                                                                      |
+| [adservice](../src/adservice/README.md)                         | Java          | Provides text ads based on given context words.                                                                                   |
+| [featureflagservice](../src/featureflagservice/README.md)         | Erlang/Elixir | CRUD feature flag service to demonstrate various scenarios like fault injection & how to emit telemetry from a feature flag reliant service.                                             |
```
Comment (reyang) [inline docs/service_table.md:15]: Consider aligning the columns.

File: docs/trace_service_features.md [hunk 3]
```diff
@@ -2,34 +2,38 @@
...
-- [Product Catalog Service](#product-catalog-service)
-- [Recommendation Service](#recommendation-service)
-- [Shipping Service](#shipping-service)
-
-## OpenTelemetry Feature Set by Services
+- [C++](#c)
+- [Erlang/Elxir](#erlangelixir)
+- [Go](#go)
+- [Java](#java)
+- [.NET](#net)
```
Comment (reyang) [inline docs/trace_service_features.md:9]: Perhaps put this on top (`.` < `A`), or consider using `C#`

**Review by reyang — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -122,7 +122,7 @@ your backend as well.
 
 ## Architecture
 
-**Online Boutique** is composed of 10 microservices written in different
+**Online Boutique** is composed of 11 microservices written in different
```
Comment (reyang) [inline README.md:125]: Maybe just use "several"

File: docs/trace_service_features.md [hunk 5]
```diff
@@ -2,34 +2,38 @@
...
-- [Email Service](#email-service)
-- [Feature Flag Service](#feature-flag-service)
-- [Frontend Service](#frontend-service)
-- [Payment Service](#payment-service)
-- [Product Catalog Service](#product-catalog-service)
-- [Recommendation Service](#recommendation-service)
-- [Shipping Service](#shipping-service)
-
-## OpenTelemetry Feature Set by Services
+- [C++](#c)
```
Comment (fatsheep9146) [inline docs/trace_service_features.md:5]: c -> c++ ?

File: docs/trace_service_features.md [hunk 6]
```diff
@@ -2,34 +2,38 @@
...
+- [Erlang/Elxir](#erlangelixir)
+- [Go](#go)
+- [Java](#java)
+- [.NET](#net)
+- [Node.js](#nodejs)
+- [Python](#python)
+- [Ruby](#ruby)
+- [Rust](#rust)
```
Comment (fatsheep9146) [inline docs/trace_service_features.md:13]: c# is missed

File: docs/trace_service_features.md (hunk 5 above)
Comment (cartersocha) [reply docs/trace_service_features.md:5]: C++ is a broken link. It needs to be c for it work

File: docs/trace_service_features.md (hunk 6 above)
Comment (cartersocha) [reply docs/trace_service_features.md:13]: .net should be above node

File: docs/trace_service_features.md (hunk 3 above)
Comment (cartersocha) [reply docs/trace_service_features.md:9]: If we use c# I think it’ll get confused with c++ since linking ignores special characters

Comment (reyang) [reply docs/trace_service_features.md:9]: Maybe `#c-1`? (it's hard to maintain though)

File: docs/trace_service_features.md (hunk 6 above)
Comment (fatsheep9146) [reply docs/trace_service_features.md:13]: got it

File: docs/service_table.md (hunk 2 above)
Comment (cartersocha) [reply docs/service_table.md]: Just updated it a bit using a vs code extension. Better but not 100% perfect

File: docs/trace_service_features.md (hunk 3 above)
Comment (cartersocha) [reply docs/trace_service_features.md:9]: Could you commit an example of what you mean on the "." or point to an example

Comment (cartersocha) [reply docs/trace_service_features.md:9]: Or we could use the trusty dotnet & cpp naming

Comment (reyang) [reply docs/trace_service_features.md:9]: > Could you commit an example of what you mean on the "." or point to an example

I guess most developers would treat alphabetical ordering as "ordering based on the ASCII code or UNICODE ordinal".
So the order should be:
* .NET
* Algol
* Basic
* C++
* D
* Erlang
* Fortran
* Go

Comment (reyang) [reply docs/trace_service_features.md:9]: In addition (I might be pedantic), I found it confusing when people mix up programming language and runtime. For example, I really think ECMAScript https://en.wikipedia.org/wiki/ECMAScript is the programming language (and the proverb is "JavaScript", which can be used to refer to ECMAScript or the very original language developed by Netscape, JScript developed by Microsoft or many other things).

I guess it's probably fine to use JavaScript as the name of the programming language, most developers would not pick whether it's ECMAScript or TypeScript.

I do think Node.js is NOT a programming language, it's a runtime.

Comment (cartersocha) [reply docs/trace_service_features.md]: updated to dotnet and javascript

File: docs/trace_service_features.md (hunk 5 above)
Comment (mic-max) [reply docs/trace_service_features.md:5]: or `cpp`

**Pull request merged**